### PR TITLE
Implement C header code generation

### PR DESCRIPTION
**Pull request by shadaj**

### CODE REVIEW

File: example/c/api.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+typedef struct DiplomatWriteable {
```
Comment (Manishearth) [inline example/c/api.h:8]: suggestion: Ideally this stuff goes in a separate file, like diplomat.h

Comment (shadaj) [reply example/c/api.h:8]: should we automatically write out the file? or just have users copy it over from the repo?

Comment (Manishearth) [reply example/c/api.h:8]: Unclear! I think probably automatically writing it out is the best, and we can provide tricks for namespacing/etc if necessary

**Review by Manishearth — APPROVED** [review]
looks like it's easier when you don't have to generate conversions :smile:

**Pull request merged**